File: app/src/main/java/com/example/homestay/domain/BookingRules.java
package com.example.homestay.domain;

public final class BookingRules {
  public static final long MIN_ADVANCE_MS = 60L * 60 * 1000;
  private BookingRules() {}

  public static String validate(
      long checkInDate,
      long checkOutDate,
      int guestCount,
      int maxGuests,
      int occupiedSlots,
      int maxSlots,
      long now) {
    if (checkInDate < now + MIN_ADVANCE_MS)
      return "Cần đặt phòng trước giờ nhận ít nhất 1 giờ";
    if (checkOutDate <= checkInDate) return "Ngày trả phòng phải sau ngày nhận phòng";
    if (guestCount < 1 || guestCount > maxGuests) return "Số khách tối đa là " + maxGuests;
    if (occupiedSlots >= maxSlots) return "Phòng đã hết chỗ trong khoảng ngày đã chọn";
    return null;
  }
}

File: app/src/main/java/com/example/homestay/domain/BookingRules.kt
package com.example.homestay.domain

object BookingRules {
    fun validate(
        checkInDate: Long,
        checkOutDate: Long,
        guestCount: Int,
        maxGuests: Int,
        occupiedSlots: Int,
        maxSlots: Int,
        todayStart: Long
    ): String? = when {
        checkInDate < todayStart -> "Ngày nhận phòng không được ở trong quá khứ"
        checkOutDate <= checkInDate -> "Ngày trả phòng phải sau ngày nhận phòng"
        guestCount !in 1..maxGuests -> "Số khách tối đa là $maxGuests"
        occupiedSlots >= maxSlots -> "Phòng đã hết chỗ trong khoảng ngày đã chọn"
        else -> null
    }
}
